python_modules/dagster/dagster/_core/execution/plan/active.py

```diff
@@ -44,17 +46,23 @@ def _default_sort_key(step: ExecutionStep) -> float:
     return int(step.tags.get(PRIORITY_TAG, 0)) * -1
 
 
+BLOCKED_INTERVAL = 1
+
+
 class ActiveExecution:
     """State machine used to track progress through execution of an ExecutionPlan."""
 
     def __init__(
         self,
+        instance: DagsterInstance,
         execution_plan: ExecutionPlan,
         retry_mode: RetryMode,
         sort_key_fn: Optional[Callable[[ExecutionStep], float]] = None,
         max_concurrent: Optional[int] = None,
         tag_concurrency_limits: Optional[List[Dict[str, Any]]] = None,
+        run_id: Optional[str] = None,
```

Comment:
Might be good to make A Thing we pass in here that handles the global concurrency interactions and have a different variant you can pass in when things are disabled or in the delegated case where the claims were already acquired. That would allow us to put The Thing under test as well as pass a mock in to ActiveExecution to verify its interactions with it.